### PR TITLE
[202405][FC] remove FC delay status field

### PR DESCRIPTION
**Pull request by stepanblyschak**
Related to https://github.com/sonic-net/sonic-swss/pull/3326.

#### Why I did it

Simplify approach to delaying counters on warm boot and fast boot. Removed FLEX_COUNTER_DELAY_STATUS_FIELD and instead postpone all FC processing to happen after apply view to not delay data plane configuration.

The CONFIG_DB should not be updated in runtime anymore for counters to be delayed.

To address https://github.com/sonic-net/sonic-buildimage/issues/20302.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Removed FLEX_COUNTER_DELAY_STATUS_FIELD set in enable_counters.py.

#### How to verify it

Run warm-boot - make sure FC orch runs only after APPLY_VIEW.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang [hunk 1]
```diff
@@ -50,9 +50,6 @@ module sonic-flex_counter {
                 leaf FLEX_COUNTER_STATUS {
                     type flex_status;
                 }
-                leaf FLEX_COUNTER_DELAY_STATUS {
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-flex_counter.yang:53]: Can we put off the YANG model change? I am wondering if there is gap between the config change and yang model change, which will cause GCU failure in PROD

**Comment by bingwang-ms** [comment]
@wen587 Can you please review again?

**Review by wen587 — APPROVED** [review]

**Comment by wen587** [comment]
Added comment in master PR: https://github.com/sonic-net/sonic-buildimage/pull/20506

**Comment by bingwang-ms** [comment]
@stepanblyschak Can you address the comment?

**Comment by bingwang-ms** [comment]
@wen587 Please provide an ADO so that I can merge it to 202405.

**Comment by wen587** [comment]
> @wen587 Please provide an ADO so that I can merge it to 202405.

Hi @bingwang-ms , please check ADO#30260883

**Review by bingwang-ms — APPROVED** [review]

**Comment by bingwang-ms** [comment]
Reverting this change since another PR https://github.com/sonic-net/sonic-buildimage/pull/20555 has been reverted

**Pull request merged**